vector/src/main/java/im/vector/app/features/login/LoginViewModel.kt

```diff
@@ -417,8 +412,8 @@ class LoginViewModel @AssistedInject constructor(
         }
 
         when (action.signMode) {
-            SignMode.SignUp             -> startRegistrationFlow()
-            SignMode.SignIn             -> startAuthenticationFlow()
+            SignMode.SignUp             -> _viewEvents.post(LoginViewEvents.OnSignModeSelected(SignMode.SignUp))
+            SignMode.SignIn             -> _viewEvents.post(LoginViewEvents.OnSignModeSelected(SignMode.SignIn))
```

Comment:
same question, Are you sure we could not go through the private method startAuthenticationFlow()?